contrib/mesos/pkg/executor/service/service.go

```diff
@@ -288,8 +381,20 @@ func (ks *KubeletExecutorServer) createAndInitKubelet(
 		MaxContainers:      kc.MaxContainerCount,
 	}
 
+	// create main pod source
 	pc := kconfig.NewPodConfig(kconfig.PodConfigNotificationIncremental, kc.Recorder)
 	updates := pc.Channel(MESOS_CFG_SOURCE)
+	go func() {
+		for u := range execUpdates {
```

Comment:
I have removed executor.done, made it private as executor.terminate and instead react on the updatesChan to close to shutdown the kubelet. Feels cleaner now.